### PR TITLE
Added default command for LEDSubsystem

### PR DESCRIPTION
**Pull request by Jetblackdragon**
Added command that turns on a blue light when note is in the index and a green light when the note is aligned.

### CODE REVIEW

File: src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java [hunk 1]
```diff
@@ -178,6 +180,18 @@ public void setAngleSpeed(double Speed) {
...
+	public boolean atTargetSpeed() {
+
+		return MathUtil.isNear(
+				setLauncherSpeedEntry.getDouble(0),
+				launcherTopEncoder.getVelocity(),
+				FLYWHEEL_VELOCITY_TOLERANCE);
+		// if(MathUtil.isNear(setLauncherSpeedEntry.getDouble(0), launcherTopEncoder.getVelocity(), )){
+		// 	return true;
+		// }
+		// return false;
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java]: Please cleanup / remove commented code that is no longer used.

File: src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        if (intakeSubsystem.getIndexSensor()) {
+            LEDSubsystem.setBLUE_LED();
+        } else if (launcherSubsystem.atTargetSpeed()) {
+            LEDSubsystem.setGREEN_LED();
+        }
+    }
+
+	@Override
+	public boolean isFinished() {
+        LEDSubsystem.setRED_LED();
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java]: Looks to be more indentation issues.  These are often driven when using tabs vs spaces.

File: src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+public class LightsCommand extends Command {
+    private final LEDSubsystem LEDSubsystem;
+    private final IntakeSubsystem intakeSubsystem;
+    private final LauncherSubsystem launcherSubsystem;
+
+    public LightsCommand(
+            LEDSubsystem LEDSubsystem, 
+            IntakeSubsystem intakeSubsystem, 
+            LauncherSubsystem launcherSubsystem) {
+		this.LEDSubsystem = LEDSubsystem;
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java]: Make sure spotless was run to properly format code. Looks to be indentation issues in this block.

File: src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java [hunk 4]
```diff
@@ -178,6 +180,18 @@ public void setAngleSpeed(double Speed) {
 		launcherAnglePIDController.setReference(Speed, ControlType.kPosition);
 	}
 
+	public boolean atTargetSpeed() {
+
+		return MathUtil.isNear(
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java]: With the ability to run independently, seems like it should be checking both motor speeds, especially for shooting into amp or just shot into amp and now shooting into speaker?
Is that the right tolerance, 0.1?

**Review by bobtseattle — CHANGES_REQUESTED** [review]
Some cleanup, and making sure logic / tolerance is what we need it to be?

**Review by bobtseattle — DISMISSED** [review]
LGTM.  We can add other color options in a different PR.

File: src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+		if (intakeSubsystem.getIndexSensor()) {
+			ledSubsystem.setBLUE_LED();
+		} else if (launcherSubsystem.isAtSpeed()) {
+			ledSubsystem.setGREEN_LED();
+		}
+	}
+
+	@Override
+	public boolean isFinished() {
+		ledSubsystem.setRED_LED();
```
Comment (TAKBS2412) [inline src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java]: Why are we setting the leds to red in `isFinished()`?

**Review by TAKBS2412 — CHANGES_REQUESTED** [review]

**Comment by kirbt** [comment]
It might be useful to add an LED color for communicating to the drivers that the robot's intake is off

**Review by jamesdooley4 — CHANGES_REQUESTED** [review]
LEDs aren't planned to be installed on the robot this week, so it'd be best to hold off on this until after the Glacier Peak competition.

**Comment by iamawesomecat** [comment]
> LEDs aren't planned to be installed on the robot this week, so it'd be best to hold off on this until after the Glacier Peak competition.

Do we want to close it until then?

**Comment by TAKBS2412** [comment]
> > LEDs aren't planned to be installed on the robot this week, so it'd be best to hold off on this until after the Glacier Peak competition.
> 
> Do we want to close it until then?

I've converted it into a draft for now.

**Review by TAKBS2412 — DISMISSED** [review]

File: src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java [hunk 6]
```diff
@@ -241,6 +241,13 @@ public boolean getSensorOverride() {
 		return sensorOverride.getBoolean(false);
 	}
 
+	public boolean isIntakeOn() {
+		return (intakeMotorFront.get() != 0
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java]: Just want to verify this logic is correct.  I may be wrong (I am assuming we are giving drive team feedback that intake can intake a note), but seems like if intake is in reject mode (assuming < 0), that we shouldn't be lighting up the intake on signal.

File: src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	public void execute() {
+		if (launcherSubsystem.isAtSpeed()) { // Checks if launcher is ready
+			ledSubsystem.setGREEN_LED();
+		} else if (intakeSubsystem.indexSensorHasNote()) { // Checks if note is in feeder
+			ledSubsystem.setBLUE_LED();
+		} else if (intakeSubsystem.isIntakeOn()) { // Checks if intake is on
+			ledSubsystem.setYELLOW_LED();
+		} else { // Everything else including intake off
+			ledSubsystem.setRED_LED();
+		}
```
Comment (bobtseattle) [inline src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java]: Is this logic correct?  Should it be green if it is up to speed but does not have a note?
Should there be a color difference between a note just entering index and a note in position to be launched?

**Review by bobtseattle — COMMENTED** [review]
It would be good to make sure there is ambiguity between the state of robot and LEDs.

File: src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java (hunk 7 above)
Comment (Jetblackdragon) [reply src/main/java/frc/team2412/robot/commands/LED/LightsCommand.java]: To be green it has been updated to check if it has a note as well. New color violet has been added to check if the feeder sensor detects a note.

File: src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java (hunk 6 above)
Comment (Jetblackdragon) [reply src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java]: Drive team told me to leave rejecting to be yellow as well.

**Review by bobtseattle — DISMISSED** [review]
LGTM - We can deal with back intake during testing.  Though, should deal with failing formatting failures.

File: src/main/java/frc/team2412/robot/Robot.java [hunk 8]
```diff
@@ -43,6 +43,9 @@ public static Robot getInstance() {
 	// Really dangerous to keep this enabled as it disables all other controls, use with caution
 	private static final boolean sysIdMode = false;
 
+	// Turns on an off auto logic
+	private static final boolean autoEnabled = false;
```
Comment (jamesdooley4) [inline src/main/java/frc/team2412/robot/Robot.java]: Likely should turn autonomous on by default

**Review by jamesdooley4 — CHANGES_REQUESTED** [review]
Need to re-enable auto

**Review by jamesdooley4 — APPROVED** [review]

**Pull request merged**